specification/cognitiveservices/data-plane/AzureOpenAI/authoring/preview/2023-04-01-preview/images.json

```diff
@@ -0,0 +1,348 @@
+            }
+         }
+      },
+      "OperationResult": {
+         "type": "array",
+         "description": "The result data of the operation, if successful",
+         "items": {
+            "$ref": "#/definitions/ImageResult"
+         }
+      },
+      "OperationResponse": {
+         "type": "object",
+         "properties": {
+            "id": {
+               "type": "string",
+               "description": "The ID of the operation.",
+               "example": "59c7e2ea-7450-4e85-9762-55c78727f4c7"
+            },
+            "created": {
+               "type": "integer",
+               "format": "int64",
+               "description": "The unix timestamp when the operation was created.",
+               "example": "1676540381"
+            },
+            "last_action": {
+               "type": "integer",
+               "format": "int64",
+               "description": "The unix timestamp for when the current state was entered.",
+               "example": "1676540381"
+            },
+            "expires_at": {
+               "type": "integer",
+               "format": "int64",
+               "description": "The unix timestamp when the operation expires.",
+               "example": "1676540381"
+            },
+            "data": {
+               "$ref": "#/definitions/OperationResult"
+            },
+            "status": {
+               "$ref": "#/definitions/Status"
+            },
+            "error": {
+               "$ref": "#/definitions/Error"
+            }
+         },
+         "required": [
+            "id",
+            "status",
+            "created",
+            "last_action"
+         ]
+      },
+      "ImageResolution": {
+         "description": "The resolution of the image (256x256, 512x512, or 1024x1024). Smaller images are faster to generate.",
+         "type": "string",
+         "enum": [
+            "256x256",
+            "512x512",
+            "1024x1024"
+         ],
+         "default": "1024x1024"
+      },
+      "GenerateImagesPostRequest": {
+         "type": "object",
+         "properties": {
+            "prompt": {
+               "type": "string",
+               "description": "The prompt to generate an image from.",
+               "minLength": 1,
+               "maxLength": 1000
+            },
+            "n": {
+               "type": "integer",
+               "description": "The number of images to generate.",
+               "minimum": 1,
+               "default": 1
+            },
+            "size": {
+               "$ref": "#/definitions/ImageResolution"
+            }
+         },
+         "required": [
+            "prompt"
+         ]
+      },
+      "GenerateImagesPostResponse": {
+         "type": "object",
+         "properties": {
+            "status": {
+               "$ref": "#/definitions/Status"
+            }
+         },
+         "required": [
+            "id"
+         ]
+      }
+   },
+   "paths": {
+      "/images/generate": {
```

Comment:
>/ [](http://example.com/codeflow?start=14&length=1)

separate with :

Reply:
We actually had it implemented like this a few weeks before, but then we decided to make sure the we are more aligned with the OpenAI API. Do you think we should still change it?